pyluna-pathology/luna/pathology/cli/generate_tiles.py

```diff
@@ -0,0 +1,151 @@
+# General imports
+import os, json, logging, yaml, sys
+import click
+
+from luna.common.custom_logger   import init_logger
+
+init_logger()
+logger = logging.getLogger('generate_tiles')
+
+from luna.common.utils import cli_runner
+
+_params_ = [('input_slide_image', str), ('output_dir', str), ('tile_size', int), ('batch_size', int), ('requested_magnification', float), ('num_cores', int)]
+
+@click.command()
+@click.argument('input_slide_image', nargs=1)
+@click.option('-o', '--output_dir', required=False,
+              help='path to output directory to save results')
+@click.option('-nc', '--num_cores', required=False,
+              help="Number of cores to use", default=4)  
+@click.option('-rts', '--tile_size', required=False,
+              help="Number of cores to use")
+@click.option('-rmg', '--requested_magnification', required=False,
+              help="Number of cores to use")
+@click.option('-bx', '--batch_size', required=False,
+              help="batch size used for inference speedup", default=64)    
+@click.option('-m', '--method_param_path', required=False,
+              help='path to a metadata json/yaml file with method parameters to reproduce results')
+def cli(**cli_kwargs):
+    """Rasterize a slide into smaller tiles
+    
+    Tiles are saved in the whole-slide tiles binary format (tiles.pil), and the corresponding manifest/header file (tiles.csv) is also generated
+
+    Neccessary data for the manifest file are: 
+    address, x_coord, y_coord, full_resolution_tile_size, tile_image_binary, tile_image_length, tile_image_size_xy, and tile_image_size_mode
```

Comment:
tile_image_size_mode -> tile_image_mode, for all docstrings that have tile_image_size_mode